### PR TITLE
Add a setting for the date-format string.

### LINKED ISSUE #4: Date format in settings
Thanks for creating the extension! I guess you saw it coming - would you mind having a setting for the date format string? All my projects are named `YYYYMMDD.<Name>` and before I start renaming everything ...

So your current format would be `yyyy-MM-dd_` (to also cover the '_' in line 103).

https://github.com/kernicPanel/bitwig-randomizer/blob/main/src/main/java/com/kernicpanel/RandomizerExtension.java#L101-L103

If you like the idea, I can have a go at it and send a PR end of week.

### PR DESCRIPTION
**Pull request by ensonic**
Fixes #4

### CODE REVIEW

**Comment by kernicPanel** [comment]
Hi! Thanks for your contribution!
I'll give it a try asap.

File: src/main/java/com/kernicpanel/RandomizerExtension.java [hunk 1]
```diff
@@ -98,9 +101,9 @@ private NoArgsCallback randomName() {
 
       if (useDate.get()) {
         LocalDate dateObj = LocalDate.now();
-        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(dateFormat.get());
```
Comment (kernicPanel) [inline src/main/java/com/kernicpanel/RandomizerExtension.java:104]: If dateFormat contains something like `hgfjhgfj` we get an `IllegalArgumentException`
What about this ?
```suggestion
        DateTimeFormatter formatter;
        try {
           formatter = DateTimeFormatter.ofPattern(dateFormat.get());
        } catch (IllegalArgumentException e) {
          formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd_");
        }
```
Or maybe, we could make some check as soon as the user changes the setting, and alert him right away ?

File: src/main/java/com/kernicpanel/RandomizerExtension.java [hunk 2]
```diff
@@ -51,6 +52,8 @@ public void init() {
 
     useDate =
         host.getPreferences().getBooleanSetting("Prepend date for filename", "Random name", true);
+    dateFormat =
```
[the remark this answers is not in the document]
Comment (ensonic) [reply src/main/java/com/kernicpanel/RandomizerExtension.java:55]: I'll change it. I was trying to keep variables somewhat short, but I am aware of the java traditions :-)

File: src/main/java/com/kernicpanel/RandomizerExtension.java (hunk 1 above)
Comment (ensonic) [reply src/main/java/com/kernicpanel/RandomizerExtension.java:104]: This is definitely better. I'll need to check the API if I there is a callback when the settings were changed and If there is a way to open a dialog.

**Comment by ensonic** [comment]
Thaks for the review and please take another look.

**Comment by ensonic** [comment]
Add docs.

**Review by kernicPanel — APPROVED** [review]
Great work!
Thank for your contribution :+1:

**Pull request merged**